File: vol1/ch1/LeetCode283-1.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int k=0;
        for(int i=0;i<nums.size();i++) {
            if (nums[i]!=0) {           //将保留的元素重新插入
                nums[k]=nums[i];
                k++;
            }
        }
        for(int i=k;i<nums.size();i++)
            nums[i]=0;
    }
};

File: vol1/ch1/LeetCode283-1.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        k=0
        for i in range(0,len(nums)):
            if nums[i]!=0:           #将保留的元素重新插入
                nums[k]=nums[i]
                k+=1
        for i in range(k,len(nums)):
            nums[i]=0
